### ISSUE TITLE
Amazon Provider SCOPE Issue

### ISSUE THREAD

**Issue by matthewkinnear** [pennersr/django-allauth#3993]
When providing SCOPE = "profile", the following error is returned:
?error_description=An+unknown+scope+was+requested&state=RtLXob2d0fckilSo&error=invalid_scope#

```
SOCIALACCOUNT_PROVIDERS = {
 'amazon': {
        'APP': {
            'client_id': allauth_amazon_client_id,
            'secret': allauth_amazon_client_secret,
            'key': ''
        },
        "SCOPE": [
            "profile"
        ]
    }
}

```
Amazon's documentation confirms that this should be "profile" as the default:
https://developer.amazon.com/docs/login-with-amazon/requesting-scopes-as-essential-voluntary.html

Code containing the default :
https://github.com/pennersr/django-allauth/blob/main/allauth/socialaccount/providers/amazon/provider.py

**Comment by pennersr** [comment]
Can you double check the URL that is used to redirect the user to Amazon? There should be a `scope=` in there, and I am curious as to what the value is.

**Comment by matthewkinnear** [comment]
> Can you double check the URL that is used to redirect the user to Amazon? There should be a `scope=` in there, and I am curious as to what the value is.

Thanks for getting back to me.

Encoded:
http://www.amazon.com/ap/oa?client_id=X&redirect_uri=http%3A%2F%2Flocalhost%3A5000%2Faccounts%2Famazon%2Flogin%2Fcallback%2F&scope=profile&response_type=code&state=kUDnEpIhXPvR0I57

Decoded:
redirect_uri: http://localhost:5000/accounts/amazon/login/callback/
scope: profile
response_type: code

**Comment by matthewkinnear** [comment]
If I use the following, I can get to the login screen:
redirect_uri: http://localhost:5000/accounts/amazon/login/callback/
scope: profile:email
response_type: code

However, after the redirect back to my app, I get the following:
`Exception occured: <class 'KeyError'>, 'user_id'
> /home/X/.cache/pypoetry/virtualenvs/X-py3.10/lib/python3.10/site-packages/allauth/socialaccount/providers/amazon/provider.py(21)extract_uid()
-> return str(data["user_id"])
`

When trying to debug, I can see that 'data' contains:
`(Pdb) print(data)
{'error_index': 'UO1CTg0OP8gu9pu5bA2qmgAAAAAAAAABjBl8wBTsLPjg9IQgoO9CJkRzxN80fklLTlqLv3Hn6Wq_GpowRwFldA0eKGQknDp59WQgspEC01hLm6gk3f0Gk0wFwJhA0KoV9QGnNQkb5IzZ_KyH55rM2n__mIgC-iupG3hnFWlGgVlp7MiICtKWy9kjny_hsH7rtkC0OUuU7xNc67RqqPMbCOKLFmUFTxg2yR9LuyICvUzI-38f16GzCz8p3tS9UGSKeN0yOQ==', 'error_description': 'The request has an invalid parameter : access_token', 'error': 'invalid_token', 'request_id': '85a1696f-f31f-4e07-bb32-27610504d373'}
`

**Comment by matthewkinnear** [comment]
> If I use the following, I can get to the login screen: redirect_uri: http://localhost:5000/accounts/amazon/login/callback/ scope: profile:email response_type: code
> 
> However, after the redirect back to my app, I get the following: `Exception occured: <class 'KeyError'>, 'user_id'
> 
> > /home/X/.cache/pypoetry/virtualenvs/X-py3.10/lib/python3.10/site-packages/allauth/socialaccount/providers/amazon/provider.py(21)extract_uid()
> > -> return str(data["user_id"])
> > `
> 
> When trying to debug, I can see that 'data' contains: `(Pdb) print(data) {'error_index': 'UO1CTg0OP8gu9pu5bA2qmgAAAAAAAAABjBl8wBTsLPjg9IQgoO9CJkRzxN80fklLTlqLv3Hn6Wq_GpowRwFldA0eKGQknDp59WQgspEC01hLm6gk3f0Gk0wFwJhA0KoV9QGnNQkb5IzZ_KyH55rM2n__mIgC-iupG3hnFWlGgVlp7MiICtKWy9kjny_hsH7rtkC0OUuU7xNc67RqqPMbCOKLFmUFTxg2yR9LuyICvUzI-38f16GzCz8p3tS9UGSKeN0yOQ==', 'error_description': 'The request has an invalid parameter : access_token', 'error': 'invalid_token', 'request_id': '85a1696f-f31f-4e07-bb32-27610504d373'} `

It's worth mentioning that the response from Amazon contains a "code", "scope" and "state".

**Comment by pennersr** [comment]
Fixed via 3c836f88

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3993
